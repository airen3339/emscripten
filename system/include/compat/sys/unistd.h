#include <unistd.h>
#include <random.h> // for getentropy()
