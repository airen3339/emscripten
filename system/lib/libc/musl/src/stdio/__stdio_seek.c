#include "stdio_impl.h"

off_t __stdio_seek(FILE *f, off_t off, int whence)
{
	off_t ret;
#ifdef SYS__llseek
	off_t off_hi =
#ifndef __EMSCRIPTEN__
		off >> 32;
#else
		0;
#endif
	if (syscall(SYS__llseek, f->fd, off_hi, off, &ret, whence)<0)
		ret = -1;
#else
	ret = syscall(SYS_lseek, f->fd, off, whence);
#endif
	return ret;
}
