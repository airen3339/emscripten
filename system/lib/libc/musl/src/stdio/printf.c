#include "stdio_impl.h"
#include <stdio.h>
#include <stdarg.h>

int printf(const char *restrict fmt, ...)
{
	int ret;
	va_list ap;
	va_start(ap, fmt);
	ret = vfprintf(stdout, fmt, ap);
	va_end(ap);
	return ret;
}

// XXX EMSCRIPTEN
int iprintf(const char *restrict fmt, ...)
{
	int ret;
	va_list ap;
	va_start(ap, fmt);
	ret = vfiprintf(stdout, fmt, ap);
	va_end(ap);
	return ret;
}

extern int __small_vfprintf(FILE *restrict f, const char *restrict fmt, va_list ap);

int __small_printf(const char *restrict fmt, ...)
{
	int ret;
	va_list ap;
	va_start(ap, fmt);
	ret = __small_vfprintf(stdout, fmt, ap);
	va_end(ap);
	return ret;
}

