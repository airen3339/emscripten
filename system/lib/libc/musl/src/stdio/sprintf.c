#include "stdio_impl.h"
#include <stdio.h>
#include <stdarg.h>

int sprintf(char *restrict s, const char *restrict fmt, ...)
{
	int ret;
	va_list ap;
	va_start(ap, fmt);
	ret = vsprintf(s, fmt, ap);
	va_end(ap);
	return ret;
}

// XXX EMSCRIPTEN
int siprintf(char *restrict s, const char *restrict fmt, ...)
{
	int ret;
	va_list ap;
	va_start(ap, fmt);
	ret = vsiprintf(s, fmt, ap);
	va_end(ap);
	return ret;
}

extern int __small_vsprintf(char *restrict s, const char *restrict fmt, va_list ap);

int __small_sprintf(char *restrict s, const char *restrict fmt, ...)
{
	int ret;
	va_list ap;
	va_start(ap, fmt);
	ret = __small_vsprintf(s, fmt, ap);
	va_end(ap);
	return ret;
}

