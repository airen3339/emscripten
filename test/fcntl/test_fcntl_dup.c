#include <unistd.h>
#include <fcntl.h>
#include <stdio.h>
#include <errno.h>
#include <assert.h>

int
main() {
    int fd, dupfd, status, bytes;
    fclose(stdin);
    fd = open("a.txt", O_CREAT | O_WRONLY, 0644);
    assert(fd == 0);
    printf("fd: %d errno: %d\n", fd, errno);
    bytes = write(fd, "hi there!\n", 11);
    assert(bytes == 11);
    printf("fd: %d errno: %d bytes: %d\n", fd, errno, bytes);

    dupfd = fcntl(fd, F_DUPFD, 0);
    assert(dupfd != fd);
    printf("dupfd: %d\n", dupfd);
    status = close(dupfd);
    printf("status: %d\n", status);
    assert(status == 0);
    bytes = write(fd, "hi again!", 10);
    printf("fd: %d errno: %d bytes: %d\n", fd, errno, bytes);
    assert(bytes == 10);
    status = close(fd);
    assert(status == 0);
}
