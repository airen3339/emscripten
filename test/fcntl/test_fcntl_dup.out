fd: 0 errno: 0
fd: 0 errno: 0 bytes: 11
dupfd: 3
status: 0
fd: 0 errno: 0 bytes: 10
